src/DonationForms/resources/app/utilities/usePubSub.ts

```diff
@@ -0,0 +1,66 @@
+import {RefObject, useEffect} from 'react';
+
+/**
+ * @unreleased
+ */
+export default function usePubSub() {
```

Comment:
can you please add some comments/descriptions to this file explaining what each part is doing for our future selves 😄